client/src/cmdlfem410x.c

```diff
@@ -758,23 +747,50 @@ static int CmdEM410xClone(const char *Cmd) {
         c_parity ^= nibble;
     }
     data[7] |= c_parity << 1;
-    // print_hex_noascii_break(data, 8, 10);
+
+    PrintAndLogEx(INFO, "Encoded to %s", sprint_hex(data, sizeof(data)));
 
     clearCommandBuffer();
     PacketResponseNG resp;
 
     if (hs) {
         lf_hitag_data_t packet;
         memset(&packet, 0, sizeof(packet));
+        int page = 1;
 
-        for (size_t page = 4; page <= 5; page++) {
+        while (page != -1) {
             packet.cmd = WHTSF_PLAIN;
             packet.page = page;
-            memcpy(packet.data, &data[(page-4)*4], 4);
+
+            switch (page) {
+                case 1:
+                    packet.data[0] = 0xCA; //compatiable for 82xx, no impact on Hitag S
+                    // clk -> TTFDR1 TTFDR0
+                    // 32  -> 0x00      4 kBit/s
+                    // 16  -> 0x10      8 kBit/s
+                    // 64  -> 0x20      2 kBit/s
+                    packet.data[1] = 0x04;
+                    switch (clk) {
+                        case 32: break;
+                        case 16: packet.data[1] |= 0x10; break;
+                        case 64: packet.data[1] |= 0x20; break;
+                    }
+                    packet.data[2] = 0;
+                    packet.data[3] = 0; //TODO: keep PWDH0?
+                    page = 4;
+                    break;
+                case 4:
+                    memcpy(packet.data, &data[HITAG_BLOCK_SIZE * 0], HITAG_BLOCK_SIZE);
+                    page = 5;
+                    break;
```

Comment:
Confusing use of  page variable. 

Needs a better solution.

Reply:
Changes to `steps`, is that more clear?

Reply:
:)
a normal `i` would have been enough ,  but its clearer now. 

thanks!